core/datasource/spec/spec.go

```diff
@@ -165,6 +165,12 @@ func (s *Spec) MatchData(data common.Data) (bool, error) {
 		return false, nil
 	}
 
+	// Don't broadcast ethcall data based unless it's 'EthKey' matches
+	// (which is currently the SpecID - see comment on the datasource.common.Data struct)
+	if data.EthKey != "" && data.EthKey != string(s.id) {
```

Comment:
could you add a little unit-test in `spec_test.go`?

Reply:
sure